main/src/main/scala/sbt/Main.scala

```diff
@@ -882,6 +882,7 @@ object BuiltinCommands {
 
   def shell: Command = Command.command(Shell, Help.more(Shell, ShellDetailed)) { s0 =>
     import sbt.internal.{ ConsolePromptEvent, ConsoleUnpromptEvent }
+    SysProp.setCloseClassLoaders()
```

Comment:
Could we also provide opt-out via a ThisBuild setting? I think I've come across multiple times where this property was suggested, so it's probably a sign that it needs to be promoted as a real thing.